course05/homework01/freekpaans/string.cc

```diff
@@ -0,0 +1,76 @@
+#include "string.h"
+
+#include <algorithm>
+#include <stdexcept>
+
+String::String()
+{
+}
+
+String::String(std::string s)
+{
+	if(s.size() > MaxChars)
+	{
+		throw std::runtime_error{"String too long: " + s};
+	}
+
+	mSize = s.size();
+	copy(s.begin(), s.end(), mChars);
+}
+
+String::String(const char* b, const char* e)
+	: mSize{static_cast<size_t>(e - b)}
```

Comment:
Would be interested in how you would get rid of the `static_cast` here

Reply:
I don't think there is a way without the cast unfortunately. That being said `ptrdiff_t` is signed and it would be better to check (at least assert, why not throwing) if b > e